diff --git a/node_modules/tsdx/dist/babelPluginTsdx.js b/node_modules/tsdx/dist/babelPluginTsdx.js
index 57d9c93..418a103 100644
--- a/node_modules/tsdx/dist/babelPluginTsdx.js
+++ b/node_modules/tsdx/dist/babelPluginTsdx.js
@@ -73,10 +73,6 @@ exports.babelPluginTsdx = rollup_plugin_babel_1.default.custom(() => ({
             { name: '@babel/plugin-proposal-optional-chaining' },
             // Adds syntax support for default value using ?? operator
             { name: '@babel/plugin-proposal-nullish-coalescing-operator' },
-            {
-                name: '@babel/plugin-transform-regenerator',
-                async: false,
-            },
             {
                 name: 'babel-plugin-macros',
             },
